File: lib/models/vod_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'json_coerce.dart';

part 'vod_item.freezed.dart';
part 'vod_item.g.dart';

@freezed
abstract class VodItem with _$VodItem {
  const factory VodItem({
    @JsonKey(name: 'stream_id') required dynamic streamId,
    @JsonKey(fromJson: coerceString) @Default('') String name,
    @JsonKey(name: 'stream_icon', fromJson: coerceStringOrNull) String? streamIcon,
    @JsonKey(fromJson: coerceStringOrNull) String? cover,
    @JsonKey(name: 'container_extension', fromJson: coerceString) @Default('mp4') String containerExtension,
    @JsonKey(name: 'category_id', fromJson: coerceStringOrNull) String? categoryId,
    @JsonKey(name: 'category_name', fromJson: coerceStringOrNull) String? categoryName,
    @JsonKey(fromJson: coerceStringOrNull) String? rating,
    @JsonKey(name: 'stream_type', fromJson: coerceStringOrNull) String? streamType,
    @JsonKey(fromJson: coerceStringOrNull) String? plot,
    @JsonKey(fromJson: coerceStringOrNull) String? description,
    @JsonKey(fromJson: coerceStringOrNull) String? added,
    @JsonKey(name: 'last_modified', fromJson: coerceStringOrNull) String? lastModified,
  }) = _VodItem;

  factory VodItem.fromJson(Map<String, dynamic> json) => _$VodItemFromJson(json);
}

extension VodItemX on VodItem {
  String get id => streamId.toString();
  String get displayIcon => streamIcon ?? cover ?? '';
}

File: tvos/UniStreamTV/UniStreamTV/Models/VodItem.swift
import Foundation

/// Video-on-demand item — mirrors Flutter's `VodItem`.
struct VodItem: Identifiable, Hashable {
    let streamId: String
    var name: String
    var streamIcon: String?
    var cover: String?
    var containerExtension: String
    var categoryId: String?
    var categoryName: String?
    var rating: String?
    var streamType: String?
    var plot: String?
    var description: String?
    var added: String?
    var lastModified: String?

    var id: String { streamId }
    var displayIcon: String { streamIcon ?? cover ?? "" }

    init(json: [String: Any]) {
        streamId = "\(json["stream_id"] ?? "")"
        name = coerceString(json["name"])
        streamIcon = coerceStringOrNull(json["stream_icon"])
        cover = coerceStringOrNull(json["cover"])
        containerExtension = coerceStringOrNull(json["container_extension"]) ?? "mp4"
        categoryId = coerceStringOrNull(json["category_id"])
        categoryName = coerceStringOrNull(json["category_name"])
        rating = coerceStringOrNull(json["rating"])
        streamType = coerceStringOrNull(json["stream_type"])
        plot = coerceStringOrNull(json["plot"])
        description = coerceStringOrNull(json["description"])
        added = coerceStringOrNull(json["added"])
        lastModified = coerceStringOrNull(json["last_modified"])
    }
}
